design_wiki/seeds/how-data-products-are-resolved.md

```
@@ -0,0 +1,43 @@
+- Edges are created only from input products which match all specified fields of the selector
+- Where the product is produced by a HOF other than a provider the relevant fields are: Concept, Creator name, and suffix
+- Where the product is produced by a provider, the relevant fields are: Concept, creator name, suffix, and layer name
+- Where the product is produced by an *explicit* provider, the layer name is also used at graph creation time
+  - Explicit providers statically specify the layer path associated with their output
+
+### Layers Checking / Resolution
+- It is desirable that a CHOF be able to take a given input product from different layers on different iterations
+  - On each iteration the output layer of the CHOF is determined by the layers of its inputs
+- Therefore layer name checking must be delayed til runtime, when data flows along the created edges
+
+#### Contentious Part
```

markdownlint-cli2 (0.23.0)

[warning] Headings should be surrounded by blank lines
Expected: 1; Actual: 0; Below

(MD022, blanks-around-headings)